diff --git a/drivers/usb/sunxi_usb/udc/sw_udc.c b/drivers/usb/sunxi_usb/udc/sw_udc.c
index 283ae4c..1e7a25e 100644
--- a/drivers/usb/sunxi_usb/udc/sw_udc.c
+++ b/drivers/usb/sunxi_usb/udc/sw_udc.c
@@ -2757,12 +2757,12 @@ static int sw_udc_set_pullup(struct sw_udc *udc, int is_on)
 {
 	DMSG_DBG_UDC("sw_udc_set_pullup\n");
 
-	is_udc_enable = is_on;
 	if(!is_peripheral_active()){
 		DMSG_PANIC("ERR: usb device is not active\n");
 		return 0;
 	}
 
+	is_udc_enable = is_on;
 
 	if(is_on){
 		sw_udc_enable(udc);
