diff -rup1 ubs_org/board/sunxi/board.c ubs_patched/board/sunxi/board.c
--- ubs_org/board/sunxi/board.c	2014-04-08 20:38:00.047921359 +0200
+++ ubs_patched/board/sunxi/board.c	2014-04-08 20:42:31.555916572 +0200
@@ -54,4 +54,8 @@ int checkboard(void)
 {
-	printf("Board: %s\n", CONFIG_SYS_BOARD_NAME);
-
+    /* RED-Brick-OPT: Changing the board name to display */
+    /*
+    printf("Board: %s\n", CONFIG_SYS_BOARD_NAME);
+    */
+    printf("Board: %s\n", "RED-Brick");
+    
 	return 0;
diff -rup1 ubs_org/boards.cfg ubs_patched/boards.cfg
--- ubs_org/boards.cfg	2014-04-08 20:38:00.063921357 +0200
+++ ubs_patched/boards.cfg	2014-04-08 20:42:31.559916572 +0200
@@ -370,3 +370,4 @@ Active  arm         armv7          sunxi
 Active  arm         armv7          sunxi       -               sunxi               A10-OLinuXino-Lime                   sun4i:A10_OLINUXINO_L,STATUSLED=226,SPL,SUNXI_EMAC                                                                                -
-Active  arm         armv7          sunxi       -               sunxi               A10s-OLinuXino-M                     sun5i:A10S_OLINUXINO_M,STATUSLED=131,AXP152_POWER,CONS_INDEX=1,SPL,SUNXI_EMAC                                                     -
+#Active  arm         armv7          sunxi       -               sunxi               A10s-OLinuXino-M                     sun5i:A10S_OLINUXINO_M,STATUSLED=131,AXP152_POWER,CONS_INDEX=1,SPL,SUNXI_EMAC                                                     -
+Active  arm         armv7          sunxi       -               sunxi               A10s-OLinuXino-M                     sun5i:A10S_OLINUXINO_M,CONS_INDEX=1,SPL                                                     -
 Active  arm         armv7          sunxi       -               sunxi               A10s-OLinuXino-M_FEL                 sun5i:A10S_OLINUXINO_M,STATUSLED=131,AXP152_POWER,CONS_INDEX=1,SPL_FEL,SUNXI_EMAC                                                 -
diff -rup1 ubs_org/common/spl/spl.c ubs_patched/common/spl/spl.c
--- ubs_org/common/spl/spl.c	2014-04-08 20:38:00.083921357 +0200
+++ ubs_patched/common/spl/spl.c	2014-04-08 20:42:31.559916572 +0200
@@ -47,5 +47,8 @@ __weak int spl_start_uboot(void)
 {
+    /* RED-Brick-OPT: Disable SPL output */
+    /*
 	puts("SPL: Please implement spl_start_uboot() for your board\n");
 	puts("SPL: Direct Linux boot not active!\n");
-	return 1;
+	*/
+    return 1;
 }
diff -rup1 ubs_org/include/configs/sun5i.h ubs_patched/include/configs/sun5i.h
--- ubs_org/include/configs/sun5i.h	2014-04-08 20:38:00.299921353 +0200
+++ ubs_patched/include/configs/sun5i.h	2014-04-08 20:42:31.559916572 +0200
@@ -14,4 +14,8 @@
 #define CONFIG_SUN5I		/* sun5i SoC generation */
-
+/* RED-Brick-OPT: Changing U-Boot prompt */
+/*
 #define CONFIG_SYS_PROMPT		"sun5i# "
+*/
+#define CONFIG_SYS_PROMPT		"RED-Brick# "
+
 #define CONFIG_MACH_TYPE		4138
diff -rup1 ubs_org/include/configs/sunxi-common.h ubs_patched/include/configs/sunxi-common.h
--- ubs_org/include/configs/sunxi-common.h	2014-04-08 20:38:00.299921353 +0200
+++ ubs_patched/include/configs/sunxi-common.h	2014-04-08 20:43:30.163915540 +0200
@@ -64,7 +64,9 @@
 #endif
+
+/* RED-Brick-OPT: Disable NAND support */
+/*
 #if 0
-/* Nand config */
 #define CONFIG_NAND
 #define CONFIG_NAND_SUNXI
-#define CONFIG_CMD_NAND                         /* NAND support */
+#define CONFIG_CMD_NAND
 #define CONFIG_SYS_MAX_NAND_DEVICE      1
@@ -72,5 +74,9 @@
 #endif
+*/
 
+/* RED-Brick-OPT: Disable memory manupulation commands */
+/*
 #define CONFIG_CMD_MEMORY
 #define CONFIG_CMD_SETEXPR
+*/
 
@@ -78,4 +84,11 @@
 #define CONFIG_CMDLINE_TAG
+/* RED-Brick-OPT: Disable INITRD TAG */
+/*
 #define CONFIG_INITRD_TAG
+*/
+
+/* RED-Brick-OPT: Disable command line history and editing */
+/*
 #define CONFIG_CMDLINE_EDITING
+*/
 
@@ -99,4 +112,6 @@
 
-/* Flat Device Tree (FDT/DT) support */
+/* RED-Brick-OPT: Disable Flat Device Tree (FDT/DT) support */
+/*
 #define CONFIG_OF_LIBFDT
+*/
 
@@ -105,5 +120,9 @@
  */
-#define CONFIG_SYS_LONGHELP	/* undef to save memory */
-#define CONFIG_SYS_HUSH_PARSER	/* use "hush" command parser    */
+ 
+/* RED-Brick-OPT: Disable U-Boot console features */
+/*
+#define CONFIG_SYS_LONGHELP
+#define CONFIG_SYS_HUSH_PARSER
 #define CONFIG_CMD_ECHO
+*/
 #define CONFIG_SYS_CBSIZE	256	/* Console I/O Buffer Size */
@@ -117,4 +136,6 @@
 
-/* standalone support */
+/* RED-Brick-OPT: Disable stand alone support */
+/*
 #define CONFIG_STANDALONE_LOAD_ADDR	0x50000000
+*/
 
@@ -129,3 +150,2 @@
 /* FLASH and environment organization */
-
 #define CONFIG_SYS_NO_FLASH
@@ -133,3 +153,3 @@
 #define CONFIG_SYS_MONITOR_LEN		(512 << 10)	/* 512 KiB */
-#define CONFIG_IDENT_STRING		" Allwinner Technology"
+#define CONFIG_IDENT_STRING		" Tinkerforge GmbH"
 
@@ -138,2 +158,4 @@
 
+/* RED-Brick-OPT: Not compiling for FEL support */
+/*
 #ifdef CONFIG_SPL_FEL
@@ -143,20 +165,15 @@
 #endif
+*/
 
+/* RED-Brick-OPT: Stripping down U-Boot environment */
+#define CONFIG_BOOTDELAY 3
+#define CONFIG_BOOTARGS \
+    "console=ttyS0,115200 root=/dev/mmcblk0p2 init=/sbin/init loglevel=0 panic=0" \
+    ""
 #define CONFIG_BOOTCOMMAND \
-	RUN_BOOT_RAM \
-	"if run loadbootenv; then " \
-	  "echo Loaded environment from ${bootenv};" \
-	  "env import -t ${scriptaddr} ${filesize};" \
-	"fi;" \
-	"if test -n \\\"${uenvcmd}\\\"; then " \
-	  "echo Running uenvcmd ...;" \
-	  "run uenvcmd;" \
-	"fi;" \
-	"if run loadbootscr; then "\
-	  "echo Jumping to ${bootscr};" \
-	  "source ${scriptaddr};" \
-	"fi;" \
-	"run autoboot;" \
-	""
+    "ext2load mmc 0:1 0x43000000 script.bin; ext2load mmc 0:1 0x48000000 uImage; bootm 0x48000000" \
+    ""
 
+/* RED-Brick-OPT: No watchdog support */
+/*
 #ifdef CONFIG_CMD_WATCHDOG
@@ -166,88 +183,9 @@
 #endif
+*/
 
-#define CONFIG_EXTRA_ENV_SETTINGS \
-	"bootm_size=0x20000000\0" \
-	"console=ttyS0,115200\0" \
-	"panicarg=panic=10\0" \
-	"extraargs=\0" \
-	"loglevel=8\0" \
-	"scriptaddr=0x44000000\0" \
-	"device=mmc\0" \
-	"partition=0:1\0" \
-	"setargs=" \
-	  "if test -z \\\\\"$root\\\\\"; then"\
-	    " if test \\\\\"$bootpath\\\\\" = \"/boot/\"; then"\
-	      " root=\"/dev/mmcblk0p1 rootwait\";"\
-	    " else" \
-	      " root=\"/dev/mmcblk0p2 rootwait\";"\
-	    " fi;"\
-	  " fi;"\
-	  " setenv bootargs console=${console} root=${root}" \
-	  " loglevel=${loglevel} ${panicarg} ${extraargs}" \
-	  "\0" \
-	"kernel=uImage\0" \
-	"bootenv=uEnv.txt\0" \
-	"bootscr=boot.scr\0" \
-	"script=script.bin\0" \
-	"loadbootscr=" \
-	  "fatload $device $partition $scriptaddr ${bootscr}" \
-	  " || " \
-	  "ext2load $device $partition $scriptaddr boot/${bootscr}" \
-	  " ||" \
-	  "ext2load $device $partition $scriptaddr ${bootscr}" \
-	  "\0" \
-	"loadbootenv=" \
-	  "fatload $device $partition $scriptaddr ${bootenv}" \
-	  " || " \
-	  "ext2load $device $partition $scriptaddr boot/${bootenv}" \
-	  " || " \
-	  "ext2load $device $partition $scriptaddr ${bootenv}" \
-	  "\0" \
-	"loadkernel=" \
-	  "if "\
-	    "bootpath=/boot/" \
-	    " && " \
-	    "ext2load $device $partition 0x43000000 ${bootpath}${script}" \
-	    " && " \
-	    "ext2load $device $partition 0x48000000 ${bootpath}${kernel}" \
-	  ";then true; elif " \
-	    "bootpath=/" \
-	    " && " \
-	    "fatload $device $partition 0x43000000 ${script}" \
-	    " && " \
-	    "fatload $device $partition 0x48000000 ${kernel}" \
-	  ";then true; elif " \
-	    "bootpath=/" \
-	    " && " \
-	    "ext2load $device $partition 0x43000000 ${bootpath}${script}" \
-	    " && " \
-	    "ext2load $device $partition 0x48000000 ${bootpath}${kernel}" \
-	  ";then true; else "\
-	    "false" \
-	  ";fi" \
-	  "\0" \
-	"autoboot=" \
-	  "run loadkernel" \
-	  " && " \
-	  "run setargs" \
-	  " && " \
-	  RESET_WATCHDOG \
-	  " && " \
-	  "bootm 0x48000000" \
-	  "\0" \
-	"boot_ram=" \
-	  "saved_stdout=$stdout;setenv stdout nc;"\
-	  "if iminfo 0x41000000; then" \
-	    " " RESET_WATCHDOG ";"\
-	    " setenv stdout $saved_stdout;" \
-	    " source 0x41000000;" \
-	  "else" \
-	    " setenv stdout $saved_stdout;" \
-	  "fi" \
-	  "\0" \
-	""
-
-#define CONFIG_BOOTDELAY	3
+/* RED-Brick-OPT: Disable commandline get and auto complete */
+/*
 #define CONFIG_SYS_BOOT_GET_CMDLINE
 #define CONFIG_AUTO_COMPLETE
+*/
 
@@ -255,3 +193,6 @@
 
-/* Accept zimage + raw ramdisk without mkimage headers */
+/* RED-Brick-OPT: Reducing file system and image format supports */
+#define CONFIG_DOS_PARTITION
+#define CONFIG_CMD_EXT2     /* with this we can access ext2 bootfs */
+/*
 #define CONFIG_CMD_BOOTZ
@@ -259,7 +200,6 @@
 
-#define CONFIG_DOS_PARTITION
-#define CONFIG_CMD_FAT		/* with this we can access fat bootfs */
-#define CONFIG_FAT_WRITE	/* enable write access */
-#define CONFIG_CMD_EXT2		/* with this we can access ext2 bootfs */
-#define CONFIG_CMD_EXT4		/* with this we can access ext4 bootfs */
+#define CONFIG_CMD_FAT
+#define CONFIG_FAT_WRITE
+#define CONFIG_CMD_EXT4
+*/
 
@@ -269,3 +209,7 @@
 #define CONFIG_SPL_LIBGENERIC_SUPPORT
+
+/* RED-Brick-OPT: Disabling SPL system info print */
+/*
 #define CONFIG_SPL_DISPLAY_PRINT
+*/
 
@@ -276,2 +220,4 @@
  */
+/* RED-Brick-OPT: Do not need SPL OS boot support */
+/*
 #if defined( CONFIG_SUN5I ) || defined ( CONFIG_SYS_THUMB_BUILD )
@@ -279,5 +225,7 @@
 #endif
+*/
 
+/* RED-Brick-OPT: Not compiling for FEL support */
+/*
 #ifdef CONFIG_SPL_FEL
-
 #define CONFIG_SPL
@@ -286,6 +234,7 @@
 #define CONFIG_SPL_TEXT_BASE		0x2000
-#define CONFIG_SPL_MAX_SIZE		0x4000		/* 16 KiB */
-
-#else /* CONFIG_SPL */
+#define CONFIG_SPL_MAX_SIZE		0x4000
 
+#else
+*/
+/* CONFIG_SPL */
 #define CONFIG_SPL_BSS_START_ADDR	0x50000000
@@ -307,4 +256,3 @@
 #define CONFIG_SPL_PAD_TO		32768		/* decimal for 'dd' */
-
-#endif /* CONFIG_SPL */
+/* CONFIG_SPL */
 /* end of 32 KiB in sram */
@@ -313,2 +261,4 @@
 
+/* RED-Brick-OPT: Do not need SPL OS boot support */
+/*
 #ifdef CONFIG_SPL_OS_BOOT
@@ -322,2 +272,3 @@
 #endif
+*/
 
@@ -326,4 +277,28 @@
 #undef CONFIG_CMD_NFS
-
-/* I2C */
+/* RED-Brick-OPT: Disable additional U-Boot commands enabled by config_cmd_default.h */
+#undef CONFIG_CMD_BDI
+#undef CONFIG_CMD_BOOTD
+#undef CONFIG_CMD_CONSOLE
+#undef CONFIG_CMD_ECHO
+#undef CONFIG_CMD_EDITENV
+#undef CONFIG_CMD_IMI
+#undef CONFIG_CMD_LOADB
+#undef CONFIG_CMD_LOADS
+#undef CONFIG_CMD_MEMORY
+#undef CONFIG_CMD_MISC
+#undef CONFIG_CMD_RUN
+#undef CONFIG_CMD_SAVEENV
+#undef CONFIG_CMD_SETGETDCR
+#undef CONFIG_CMD_SOURCE
+#undef CONFIG_CMD_XIMG
+
+#undef CONFIG_VERSION_VARIABLE
+#undef CONFIG_REGEX
+#undef CONFIG_CMD_GO
+#undef CONFIG_CMD_ENV_CALLBACK
+#undef CONFIG_CMD_ENV_FLAGS
+#undef CONFIG_CMD_ENV_EXISTS
+        
+/* RED-Brick-OPT: Disable I2C support used by AXP152 PMU */
+/*
 #ifndef CONFIG_SUN6I
@@ -336,14 +311,20 @@
 #define CONFIG_CMD_I2C
+*/
 
-/* Watchdog */
+/* RED-Brick-OPT: Disable watchdog support */
+/*
 #if 0
-#define CONFIG_WATCHDOG			/* automatic watchdog support */
-#define CONFIG_CMD_WATCHDOG		/* watchdog command setting the watchdog timeout */
+#define CONFIG_WATCHDOG
+#define CONFIG_CMD_WATCHDOG
 #endif
+*/
 
-/* GPIO */
+/* RED-Brick-OPT: Disable GPIO/LED support */
+/*
 #define CONFIG_SUNXI_GPIO
 #define CONFIG_CMD_GPIO
+*/
 
-/* PMU */
+/* RED-Brick-OPT: Disable AXP152 PMU support */
+/*
 #if !defined CONFIG_AXP152_POWER && !defined CONFIG_AXP221_POWER && !defined CONFIG_NO_AXP
@@ -354,3 +335,6 @@
 #endif
+*/
 
+/* RED-Brick-OPT: Disable GPIO/LED support */
+/*
 #ifdef CONFIG_STATUSLED
@@ -395,2 +379,3 @@
 #endif
+*/
 
@@ -403,5 +388,6 @@
 
-/* Ethernet support */
+/* RED-Brick-OPT: Disable ethernet support */
+/*
 #ifdef CONFIG_SUNXI_EMAC
-#define CONFIG_MII			/* MII PHY management		*/
+#define CONFIG_MII
 #define CONFIG_CMD_MII
@@ -411,7 +397,7 @@
 #ifdef CONFIG_SUNXI_GMAC
-#define CONFIG_DESIGNWARE_ETH		/* GMAC can use designware driver */
+#define CONFIG_DESIGNWARE_ETH
 #define CONFIG_DW_AUTONEG
-#define CONFIG_PHY_GIGE			/* GMAC can use gigabit PHY	*/
+#define CONFIG_PHY_GIGE
 #define CONFIG_PHY_ADDR		1
-#define CONFIG_MII			/* MII PHY management		*/
+#define CONFIG_MII
 #define CONFIG_PHYLIB
@@ -426,3 +412,3 @@
 #define CONFIG_CMD_SNTP
-#define CONFIG_TIMESTAMP		/* Needed by SNTP */
+#define CONFIG_TIMESTAMP
 #define CONFIG_CMD_DNS
@@ -445,2 +431,3 @@
 #endif
+*/
 
